### PR TITLE
Convert jshint to eslint

### PR DESCRIPTION
**Pull request by lboyette-okta**
bacon: test
Resolves: OKTA-89554

@rchild-okta 
@magizh-okta

### CODE REVIEW

File: lib/token.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* globals FRAME_ID */
-/*jshint maxcomplexity:15 */
-
+/* eslint-disable complexity */
```
Comment (rchild-okta) [inline lib/token.js]: You can add this in one line, i.e. 
```javascript
/* eslint complexity:0, max-statements:0 */
```

File: lib/util.js [hunk 2]
```diff
@@ -9,7 +9,7 @@
  *
  * See the License for the specific language governing permissions and limitations under the License.
  */
-
+/* globals Uint8Array */
```
Comment (rchild-okta) [inline lib/util.js]: Interesting - I wonder if it's better instead to a) not use es3 in our .eslintrc file or b) if possible set es5 for this method only

File: package.json [hunk 3]
```diff
@@ -15,7 +15,9 @@
     "registry": "https://artifacts.aue1d.saasure.com/artifactory/api/npm/npm-okta"
   },
   "scripts": {
-    "lint:report": "grunt jshint",
+    "lint": "eslint ./lib ./test",
+    "lint:fix": "eslint --fix ./lib ./test",
+    "lint:report": "eslint -f checkstyle -o build2/loginjs-checkstyle-result.xml ./lib ./test",
```
Comment (rchild-okta) [inline package.json]: Would be nice to use same convention as other repos, i.e. build2/reports/lint/eslint-checkstyle-result.xml

File: test/spec/oauth.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
-/* jshint es3:false */
-/* jshint maxcomplexity:8 */
 /* globals define, expect, spyOn */
+/* eslint-disable max-statements */
```
Comment (rchild-okta) [inline test/spec/oauth.js]: Same here - one line

Comment (lboyette-okta) [inline test/spec/oauth.js]: will do

File: lib/util.js (hunk 2 above)
Comment (lboyette-okta) [inline lib/util.js]: I'll remove es3 in our .eslintrc

File: lib/token.js (hunk 1 above)
Comment (lboyette-okta) [inline lib/token.js]: I think I'll use


```javascript
/* eslint-disable complexity, max-statements */
```

unless the `:0` suffix is how we're disabling it elsewhere

File: package.json (hunk 3 above)
Comment (lboyette-okta) [inline package.json]: Gotcha. I just copied it from the old Gruntfile.js, so I figured that's what we did elsewhere. I'll change it

File: package.json [hunk 5]
```diff
@@ -15,7 +15,9 @@
     "registry": "https://artifacts.aue1d.saasure.com/artifactory/api/npm/npm-okta"
   },
   "scripts": {
-    "lint:report": "grunt jshint",
+    "lint": "eslint ./lib ./test",
```
Comment (rchild-okta) [inline package.json]: Do we need to explicitly call out these directories, or can we simply do "eslint ." and then ignore any modules in an .eslintignore file?

File: package.json (hunk 3 above)
Comment (rchild-okta) [inline package.json]: Yeah we recently standardized on the build/reports once we started doing modules in the npm repo.

File: test/util/util.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
-/*jshint es3:false */
-/*jshint maxcomplexity:15 */
 /* globals jasmine, define, spyOn, expect, JSON */
+/* eslint-disable max-statements */
```
Comment (rchild-okta) [inline test/util/util.js]: combine here as well

**Comment by rchild-okta** [comment]
Overall looks good. Had some small comments, but after addressing squash and 🚚

File: package.json (hunk 5 above)
Comment (lboyette-okta) [inline package.json]: we can add the other dirs to the .eslintignore file

**Pull request merged**